src/Chapter4.hs

```diff
@@ -673,7 +710,28 @@ Specifically,
  ❃ Implement the function to convert Tree to list
 -}
 
+--"full binary tree" implementation → nodes have two or no child nodes
+data Tree a
+    = Leaf a 
+    | Node a (Tree a) (Tree a)
```

Comment:
Yes, this is the correct idea 🎄 

Alternatively you can see the following solution in the wild:

```haskell
data Tree a
    = Empty 
    | Node a (Tree a) (Tree a)
```

And the `Leaf` in this case would be just `Node a Empty Empty` 🙂